lib/ical/stringify.js

```diff
@@ -118,18 +118,20 @@ ICAL.stringify = (function() {
       if (params.hasOwnProperty(paramName)) {
         var multiValue = (paramName in designSet.param) && designSet.param[paramName].multiValue;
         if (multiValue && Array.isArray(value)) {
+          value = value.map(stringify._rfc6868Unescape);
           if (designSet.param[paramName].multiValueSeparateDQuote) {
-            multiValue = '"' + multiValue + '"';
+            value = value.map(function(v) { return '"' + v + '"'; });
           }
-          value = value.map(stringify._rfc6868Unescape);
+          value = value.map(stringify.propertyValue);
           value = stringify.multiValue(value, multiValue, "unknown", null, designSet);
```

Comment:
We are iterating `value` with `map` up to three times here, I think it might make sense to reduce this to one iteration.